internal/controller/nim/account_controller.go

```diff
@@ -213,6 +245,39 @@ func (r *AccountReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	logger.V(1).Info(runtimesOk)
 	meta.SetStatusCondition(&targetStatus.Conditions, makeAccountFailureCondition(account.Generation, runtimesOk))
 
+	// log the available model Ids
+	var size = len(availableRuntimes)
+	modelIds := make([]string, size)
+	for i := 0; i < size; i++ {
+		modelIds[i] = availableRuntimes[i].Name
+	}
+	if marshIds, marshErr := json.Marshal(modelIds); marshErr != nil {
+		logger.V(1).Error(runtimesErr, "failed to marshal the model Ids")
+		return ctrl.Result{}, marshErr
+	} else {
+		logger.V(1).Info("fetched the available NIM models", "model Ids", json.RawMessage(marshIds))
+	}
+
+	// check the selected runtimes
+	if selectedModelList, err := r.getSelectedModelList(ctx, account.Spec.ModelListConfig, account.Namespace); err != nil {
+		logger.V(1).Error(err, "failed to get the selected model list")
+		return ctrl.Result{}, err
+	} else if len(selectedModelList) > 0 {
+		if marshIds, marshErr := json.Marshal(selectedModelList); marshErr != nil {
+			logger.V(1).Error(runtimesErr, "failed to marshal the model Ids")
+			return ctrl.Result{}, marshErr
+		} else {
+			logger.V(1).Info("got the selected NIM model list", "model Ids", json.RawMessage(marshIds))
+		}
```

Comment:
If the selected models list is created from the ConfigMap, why do we need to go to such length as marshalling again just so we could log it. Do we need the content of the ConfigMap in the logs?

Reply:
I need the content in the logs, so that I can copy the data and quickly reproduce the scenario without the actual config map.

Reply:
I think this [comment](https://github.com/opendatahub-io/odh-model-controller/pull/405#discussion_r2027249674) is also relevant here.